pkg/validation/app.go

```diff
@@ -75,13 +75,36 @@ func (v *Validator) ValidateApp(ctx context.Context, app v1alpha1.App) (bool, er
 }
 
 func (v *Validator) validateCatalog(ctx context.Context, cr v1alpha1.App) error {
-	if key.CatalogName(cr) != "" {
-		_, err := v.g8sClient.ApplicationV1alpha1().AppCatalogs().Get(ctx, key.CatalogName(cr), metav1.GetOptions{})
+	var err error
+
+	if key.CatalogName(cr) == "" {
+		return nil
+	}
+
+	var namespaces []string
+	{
+		if key.CatalogNamespace(cr) != "" {
+			namespaces = []string{key.CatalogNamespace(cr)}
+		} else {
+			namespaces = []string{metav1.NamespaceDefault, "giantswarm"}
```

Comment:
We need this for the integration tests because appcatalogsync doesn't sync these namespaces.

But we need to do this at some point anyway.